AmplifyPlugins/API/Sources/AWSAPIPlugin/Configuration/AWSAPIEndpointInterceptors.swift

```diff
@@ -16,8 +16,17 @@ struct AWSAPIEndpointInterceptors {
     let apiAuthProviderFactory: APIAuthProviderFactory
     let authService: AWSAuthServiceBehavior?
 
+    /// The order of interceptor decoration is as follows:
+    /// 1. **prelude interceptors**: append default auth info
+    /// 2. **cutomize headers**: headers provided when constructing the request
+    /// 3. **customer interceptors**: customer defined interceptors
+    /// 4. **postlude interceptors**: calculate checksums, signatures
+    var preludeInterceptors: [URLRequestInterceptor] = []
```

Comment:
Thanks, I will update the comment. 

Since the interceptor is mutating the original URLRequest, the applying order is important. That's the key point of this PR. The name `prelude` and `postlude` is to emphsize that this is solely related to the sequence in which interceptors are applied. The comments following each interceptor stage are merely examples, and I plan to remove them to simplify understanding. Our main focus should be on highlighting the interceptor order.